cli/cli_app.go

```diff
@@ -389,6 +367,46 @@ func RunTerragrunt(terragruntOptions *options.TerragruntOptions) error {
 	return runTerragruntWithConfig(terragruntOptions, terragruntConfig, false)
 }
 
+// Check the version constraints of both terragrunt and terraform. Note that as a side effect this will set the
+// following settings on terragruntOptions:
+// - TerraformPath
+// - TerraformVersion
```

Comment:
NIT: Seems a bit odd to do this as a side effect. Should we have two methods, one that sets the version numbers, and one that checks constraints?

Reply:
I had the exact same thought, but then it is a bit hairy to refactor right now because `RunTerraformCommand` relies on `TerraformPath` being set on the `terragruntOptions`, and I don't want to make a copy in this function just to avoid the side effect.

Reply:
Oh wait I guess you are suggesting do the side effects in a separate function first. Hmm... I guess that is cleaner. Will fix.

Reply:
Ok on second thought, I'm going to keep as is. After looking through the code and attempting a refactor, splitting the function turned out to be awkward because I still need to reference the partially parsed terragrunt config after the side effects. So it sort of becomes three functions:

- partialParseTerragruntConfig
- setTerraformVersionInfo (side effect)
- checkVersionConstraints

This felt a bit too verbose for my taste, considering how big the main function is now.